File: hw/hw1/Parser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace hw1
{
    class Parser
    {
        private static readonly List<string> availableOperation = new List<string>{ "+", "-", "*", "/", };

        public static int TryParseValues(string[] args, out int arg1, out int arg2, out string op)
        {
            var isArg1Int = int.TryParse(args[0], out arg1);
            var isArg2Int = int.TryParse(args[2], out arg2);
            op = args[1];
            if (!isArg1Int || !isArg2Int)
            {
                Console.WriteLine($"{args[0]} {args[1]} {args[2]} Incorrect arguments ");
                return 1;
            }

            if (!availableOperation.Contains(op))
            {
                Console.WriteLine($"{args[0]} {args[1]} {args[2]} Incorrect arguments ");
                return 2;
            }

            return 0;
        }
    }
}

File: hw4/Calculator/Parser.fs
namespace Calc
module Parser =
    let isSupported (operation : string) =
        match operation with
            | "+" -> true
            | "-" -> true
            | "*" -> true
            | "/" -> true
            | _ -> false
    let TryParseValues (args : string[]) (arg1 : outref<int>) (arg2 : outref<int>) (operation : outref<string> ) =
        let isArg1Int = System.Int32.TryParse(args.[0], &arg1)
        let isArg2Int = System.Int32.TryParse(args.[2], &arg2)
        operation <- args.[1]
        if not (isArg1Int && isArg2Int) then
             printfn $"{args.[0]} {args.[1]} {args.[2]} invalid arguments."
             1
        elif not (isSupported args.[1]) then
             printf $"{args.[0]} {args.[1]} {args.[2]} invalid arguments."
             2
        else 0